src/app/features/collectibles/components/_collectible-types/collectible-iframe.tsx

```diff
@@ -0,0 +1,38 @@
+import { ReactNode, useState } from 'react';
+
+import { Iframe } from '@app/ui/components/iframe';
+
+import { CollectibleItemLayout, CollectibleItemLayoutProps } from '../collectible-item.layout';
+import { ImageUnavailable } from '../image-unavailable';
+
+interface CollectibleIframeProps extends Omit<CollectibleItemLayoutProps, 'children'> {
+  icon: ReactNode;
+  src: string;
+}
+export function CollectibleIframe({ icon, src, ...props }: CollectibleIframeProps) {
+  const [isError, setIsError] = useState(false);
+
+  if (isError)
+    return (
+      <CollectibleItemLayout collectibleTypeIcon={icon} {...props}>
+        <ImageUnavailable />
+      </CollectibleItemLayout>
+    );
+
+  return (
+    <CollectibleItemLayout collectibleTypeIcon={icon} {...props}>
+      <Iframe
+        onError={() => setIsError(true)}
+        src={src}
+        styles={{
+          aspectRatio: '1 / 1',
+          height: '100%',
+          objectFit: 'cover',
+          overflow: 'hidden',
```

Comment:
feel like overflow should be in the other one? or remove entirely

Reply:
Yeah, I think you are right.